v2/googlecloud-to-googlecloud/src/test/java/com/google/cloud/teleport/v2/templates/bigtablechangestreamstovectorsearch/VectorSearchResourceManager.java

```diff
@@ -0,0 +1,361 @@
+    endpointClient = IndexEndpointServiceClient.create(endpointSettings.build());
+
+    // All three (index, endpoint, deployment) seem to have to exist before the public domain on the
+    // endpoint is
+    // necessarily created? Sometimes it is empty if we try to retrieve it too quickly.
+    testIndex = findOrCreateIndex();
+    testEndpoint = findOrCreateEndpoint();
+    testDeployedIndex = findOrCreateDeployedIndex();
+
+    LOG.debug(
+        "Creating match client with endpoint {}",
+        testEndpoint.getPublicEndpointDomainName() + ":443");
+
+    var matchSettings =
+        MatchServiceSettings.newBuilder()
+            .setEndpoint(testEndpoint.getPublicEndpointDomainName() + ":443");
+
+    matchClient = MatchServiceClient.create(matchSettings.build());
+  }
+
+  private Index findOrCreateIndex() throws Exception {
+    LOG.debug("Doing find-or-create for test index {}", TEST_INDEX_NAME);
+    for (var i : indexClient.listIndexes(parent).iterateAll()) {
+      if (i.getDisplayName().equals(TEST_INDEX_NAME)) {
+        LOG.debug("Using existing index: {}:{}", i.getDisplayName(), i.getName());
+        return i;
+      } else {
+        LOG.debug("Ignoring index: {}:{}", i.getDisplayName(), i.getName());
+      }
+    }
+
+    LOG.debug("Index {} does not exist, creating", TEST_INDEX_NAME);
+    return createIndex(TEST_INDEX_NAME);
+  }
+
+  private IndexEndpoint findOrCreateEndpoint() throws Exception {
+    LOG.debug("Doing find-or-create for test endpoint {}", TEST_ENDPOINT_NAME);
+
+    for (var e : endpointClient.listIndexEndpoints(parent).iterateAll()) {
+      if (e.getDisplayName().equals(TEST_ENDPOINT_NAME)) {
+        LOG.debug("Using existing endpoint: {}:{}", e.getDisplayName(), e.getName());
+        return e;
+      } else {
+        LOG.debug("Ignoring endpoint {}:{}", e.getDisplayName(), e.getName());
+      }
+    }
+
+    LOG.debug("Endpoint {} does not exist, creating", TEST_ENDPOINT_NAME);
+    return createEndpoint(TEST_ENDPOINT_NAME);
+  }
+
+  private DeployedIndex findOrCreateDeployedIndex() throws Exception {
+    LOG.debug("Doing find-or-create for test index deployment {}", TEST_DEPLOYED_INDEX_NAME);
+
+    for (var d : testEndpoint.getDeployedIndexesList()) {
+      if (d.getId().equals(TEST_DEPLOYED_INDEX_NAME)) {
+        LOG.debug("Using existing deployment: {}:{}", d.getDisplayName(), d.getId());
+        return d;
+      } else {
+        LOG.debug("Ignoring deployment {}:{}", d.getDisplayName(), d.getId());
+      }
+    }
+
+    LOG.debug("DeployedIndex {} does not exist, creating", TEST_DEPLOYED_INDEX_NAME);
+    return deployIndexToEndpoint(testIndex.getName(), testEndpoint.getName());
+  }
+
+  private Index createIndex(String indexName) throws Exception {
+    // This is somewhat of a black box, copied from an index in a good state.
+    // The resulting index will have a dimensionality of 10
+    final CharSequence indexSchema =
```

Comment:
This is a little weird; As far as I could tell, most of the properties of an index are specified as part of this big metadata schema document, rather than just being exposed as properties you can set on the builder itself, not really sure why this is the case. The values I'm using here are taken from an index I created and then queried, along with the schema URI property below.